tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html

```diff
@@ -3138,17 +3144,65 @@ <h2>Create a distance feature</h2>
             thenDoFn(result);
           }
         });
-        this.requestManager.request(url, postData).then(wrapperFn);
+        this.requestManager.request(url, postData).then(wrapperFn)
+            .catch(reason => {
+              this.exampleStatusStr = 'Request failed'
+              self.showToast_('Request failed: ' + reason);
+              self.$.spinner.hidden = true;
+            });
+      },
+
+      /**
+       * Extract the data point feature value from an example feature. If
+       * squeeze is true then convert length 1 arrays into single objects.
+       */
+      dataPointValueFromFeature_: function(feature, squeeze) {
+        let v = null;
+        if (feature.int64List) {
+          v = feature.int64List.value;
+        } else if (feature.floatList) {
+          v = feature.floatList.value;
+        } else if (feature.bytesList && feature.bytesList.value) {
+          v = feature.bytesList.value.map(val => {
+            // Strings above a max length are most likely binary encoded data
+            // and not strings and therefore shouldn't be decoded for display
+            // in Facets Dive.
+            if (val.length > maxStringLengthToDecode) {
+              return 'ERROR: string not decoded due to length';
+            }
+            try {
+              return atob(val);
+            } catch (err) {
+              console.error('Error decoding string: ' + err);
+              return 'ERROR: error decoding string';
+            }
+          });
+        }
+        if (v) {
```

Comment:
For easier readability, can you use a separate and descriptive variable name for `v`? Reading between lines, it seems like `v` up to this line is regarded as string (or null) and we are returning oneOf `[null, NaN, number, string]` (string in case squeeze == false)

Reply:
clarified with vars valueList and singleValue